lib/usd/translators/shading/usdUVTextureReader.cpp

```diff
@@ -214,43 +215,171 @@ bool PxrMayaUsdUVTexture_Reader::Read(UsdMayaPrimReaderContext* context)
             // as a relationship like texture paths inside USDZ assets.
             val = SdfAssetPath(filePath);
         }
-
         // Re-fetch the file name in case it is UDIM-tagged
-        filePath = val.UncheckedGet<SdfAssetPath>().GetAssetPath();
+        std::string unresolvedFilePath = val.UncheckedGet<SdfAssetPath>().GetAssetPath();
         mayaAttr = depFn.findPlug(_tokens->fileTextureName.GetText(), true, &status);
-        if (status == MS::kSuccess) {
-
-            // Handle UDIM dexture files:
-            std::string::size_type udimPos = filePath.rfind(_tokens->UDIMTag.GetString());
-            if (udimPos != std::string::npos) {
-                MPlug tilingAttr = depFn.findPlug(_tokens->uvTilingMode.GetText(), true, &status);
-                if (status == MS::kSuccess) {
-                    tilingAttr.setInt(3);
-
-                    // USD did not resolve the path to absolute because the file name was not an
-                    // actual file on disk. We need to find the first tile to help Maya find the
-                    // other ones.
-                    std::string udimPath(filePath.substr(0, udimPos));
-                    udimPath += "1001";
-                    udimPath += filePath.substr(udimPos + _tokens->UDIMTag.GetString().size());
-
-                    Usd_Resolver res(&prim.GetPrimIndex());
-                    for (; res.IsValid(); res.NextLayer()) {
-                        std::string resolvedName
-                            = SdfComputeAssetPathRelativeToLayer(res.GetLayer(), udimPath);
-
-                        if (!resolvedName.empty() && !ArIsPackageRelativePath(resolvedName)
-                            && resolvedName != udimPath) {
-                            udimPath = resolvedName;
-                            break;
-                        }
+        if (status != MS::kSuccess) {
+            TF_RUNTIME_ERROR(
+                "Could not find the built-in attribute fileTextureName on a Maya file node: %s! "
+                "Something is seriously wrong with your current Maya session.",
+                depFn.name().asChar());
+            return false;
+        }
+        // Handle UDIM texture files:
+        std::string::size_type udimPos = unresolvedFilePath.rfind(_tokens->UDIMTag.GetString());
+        if (udimPos != std::string::npos) {
+            MPlug tilingAttr = depFn.findPlug(_tokens->uvTilingMode.GetText(), true, &status);
+            if (status == MS::kSuccess) {
+                tilingAttr.setInt(3);
+
+                // USD did not resolve the path to absolute because the file name was not an
+                // actual file on disk. We need to find the first tile to help Maya find the
+                // other ones.
+                // TODO: (yliangsiew) Why is this naming convention being hardcoded if right
+                // above "explicit tiling" option is being selected?
```

Comment:
Maya will validate the path using utilities found in:
Python37/Lib/site-packages/maya/app/general/fileTexturePathResolver.py
and these require the hardcoded convention to recognize UDIM paths even if the right option is selected for tiling.